homeassistant/components/met/coordinator.py

```diff
@@ -0,0 +1,126 @@
+"""DataUpdateCoordinator for Met.no integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from datetime import timedelta
+import logging
+from random import randrange
+from types import MappingProxyType
+from typing import Any, Self
+
+import metno
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ELEVATION,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    EVENT_CORE_CONFIG_UPDATE,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_TRACK_HOME, DOMAIN
+
+# Dedicated Home Assistant endpoint - do not change!
+URL = "https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CannotConnect(HomeAssistantError):
+    """Unable to connect to the web site."""
+
+
+class MetDataUpdateCoordinator(DataUpdateCoordinator["MetWeatherData"]):
```

Comment:
We could move `MetWeatherData` above the coordinator, so we don't need to specify `MetWeatherData` as string

Reply:
Oh yea, that makes absolute sense, thanks for this 👍